### ISSUE TITLE
Unable to get FLOPs on model with tf.nn.bidirectional_dynamic_rnn

### ISSUE THREAD

**Issue by lissyx** [tensorflow/tensorflow#17274]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Debian/Sid
- **TensorFlow installed from (source or binary)**: binary, tensorflow-gpu
- **TensorFlow version (use command below)**: 1.5.0
- **Python version**: 2.7
- **Bazel version (if compiling from source)**: N/A
- **GCC/Compiler version (if compiling from source)**: N/A
- **CUDA/cuDNN version**: CUDA 9.0/CuDNN v7
- **GPU model and memory**: GTX1080, 8GB

Running the model under `benchmark_model`, I'm unable to get any FLOPs value, and there are several strange pieces in the output:
```
2018-02-26 14:37:55.738267: W tensorflow/core/util/stat_summarizer.cc:78] Output tensor changed between runs for 'bidirectional_rnn/fw/fw/while/Switch
2018-02-26 14:37:55.738500: W tensorflow/core/util/stat_summarizer.cc:78] Output tensor changed between runs for 'bidirectional_rnn/bw/bw/while/Switch_1
2018-02-26 14:37:55.738507: W tensorflow/core/util/stat_summarizer.cc:78] Output tensor changed between runs for 'bidirectional_rnn/bw/bw/while/Switch
2018-02-26 14:37:55.739153: W tensorflow/core/util/stat_summarizer.cc:78] Output tensor changed between runs for 'bidirectional_rnn/fw/fw/while/Switch_3
2018-02-26 14:37:55.739165: W tensorflow/core/util/stat_summarizer.cc:78] Output tensor changed between runs for 'bidirectional_rnn/fw/fw/while/Switch_2
2018-02-26 14:37:55.739171: W tensorflow/core/util/stat_summarizer.cc:78] Output tensor changed between runs for 'bidirectional_rnn/fw/fw/while/Switch_4
2018-02-26 14:37:55.739300: W tensorflow/core/util/stat_summarizer.cc:78] Output tensor changed between runs for 'bidirectional_rnn/bw/bw/while/Switch_3
2018-02-26 14:37:55.739307: W tensorflow/core/util/stat_summarizer.cc:78] Output tensor changed between runs for 'bidirectional_rnn/bw/bw/while/Switch_4
2018-02-26 14:37:55.739311: W tensorflow/core/util/stat_summarizer.cc:78] Output tensor changed between runs for 'bidirectional_rnn/bw/bw/while/Switch_2
```

And
```
2018-02-26 14:38:33.921383: E tensorflow/tools/benchmark/benchmark_model.cc:579] FLOPs calculation failed with Internal: Retval[4] has already been set.
```

**Comment by lissyx** [comment]
Generate the model: `python make_rnn_test.py test_benchmark.pb`
Run the benchmark: `./bazel-bin/tensorflow/tools/benchmark/benchmark_model --graph=test_benchmark.pb.pb --input_layer="garbage_batch_x:0" --input_layer_type=float --input_layer_shape="1,1,492" --input_layer_values="0.0" --output_layer="garbage_outputs:0" --show_flops=true --max_num_runs=1000`

**Comment by lissyx** [comment]
[log run](https://github.com/tensorflow/tensorflow/files/1758801/test_benchmark.log)

**Comment by lissyx** [comment]
```
#!/usr/bin/env python
# -*- coding: utf-8 -*-
from __future__ import absolute_import, division, print_function

import tensorflow as tf
from tensorflow.python.tools import freeze_graph

import re
import os
import sys

if len(sys.argv) != 2:
    print("%s: graphName" % (sys.argv[0]))
    print("will produce: graphName.{pb,pbtxt} and graphName.dot")
    print("got:", sys.argv)
    sys.exit(1)

graph_pb   = os.path.abspath("%s.pb" % sys.argv[1])
graph_ckpt = os.path.abspath("%s.ckpt" % sys.argv[1])
temp_pb    = os.path.abspath("%s.non-frozen.pb" % sys.argv[1])
config_pb  = os.path.abspath("%s.pbtxt" % sys.argv[1])
graph_dot  = os.path.abspath("%s.dot" % sys.argv[1])

use_rnn = False

g = tf.Graph()
with g.as_default() as g:
    n_input    = 26
    n_context  = 9
    n_cell_dim = 2
    n_hidden_3 = 2 * n_cell_dim

    batch_x = tf.placeholder(tf.float32, [None, None, n_input + 2*n_input*n_context], name='garbage_batch_x')
    seq_length = tf.placeholder(tf.int32, [None], name='garbage_seq_length')

    batch_x_shape = tf.shape(batch_x, name='garbage_batch_x_shape')
    batch_x = tf.transpose(batch_x, [1, 0, 2], name='garbage_batch_x_transpose')
    batch_x = tf.reshape(batch_x, [-1, n_input + 2*n_input*n_context], name='garbage_batch_x_reshape')
    layer_input = tf.reshape(batch_x, [-1, batch_x_shape[0], n_hidden_3], name='garbage_layer_input')

    lstm_fw_cell = tf.contrib.rnn.BasicLSTMCell(n_cell_dim, forget_bias=1.0, state_is_tuple=True)
    lstm_bw_cell = tf.contrib.rnn.BasicLSTMCell(n_cell_dim, forget_bias=1.0, state_is_tuple=True)

    rnn_fw_cell = tf.contrib.rnn.BasicRNNCell(n_cell_dim)
    rnn_bw_cell = tf.contrib.rnn.BasicRNNCell(n_cell_dim)

    outputs, output_states = tf.nn.bidirectional_dynamic_rnn(cell_fw=rnn_fw_cell if use_rnn else lstm_fw_cell,
                                                             cell_bw=rnn_bw_cell if use_rnn else lstm_bw_cell,
                                                             inputs=layer_input,
                                                             dtype=tf.float32,
                                                             time_major=True,
                                                             sequence_length=None)

    outputs = tf.concat(outputs, 2, name='garbage_outputs')

session_config = tf.ConfigProto(allow_soft_placement=True, log_device_placement=False)

with tf.Session(graph=g, config=session_config) as sess:
    init_op = tf.global_variables_initializer()
    saver = tf.train.Saver()

    r = sess.run(init_op)
    #sess.run(outputs)

    tf.train.write_graph(g, os.path.dirname(temp_pb), os.path.basename(temp_pb), as_text=False)

    # Save the variables to disk.
    save_path = saver.save(sess, graph_ckpt)
    print("Model saved in path: %s" % save_path)

    # Freeze graph
    input_graph_path = temp_pb
    input_saver_def_path = ''
    input_binary = True
    output_node_names = 'garbage_outputs'
    restore_op_name = 'save/restore_all'
    filename_tensor_name = 'save/Const:0'
    output_graph_path = graph_pb
    clear_devices = False
    freeze_graph.freeze_graph(input_graph_path, input_saver_def_path,
                               input_binary, save_path, output_node_names,
                               restore_op_name, filename_tensor_name,
                               output_graph_path, clear_devices, '')
```

**Comment by lissyx** [comment]
Switching to RNN cell does not help, removing the layer `tf.nn.bidirectional_dynamic_rnn` helps.

**Comment by angersson** [comment]
This question is better asked on  [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a  bug or feature request. There is also a larger community that reads questions there. Thanks!

**Comment by lissyx** [comment]
@angersson I'm not asking now how to use `benchmark_model`, I'm reporting that a model with a `tf.nn.bidirectional_dynamic_rnn` is actually breaking (at least some) features of `benchmark_model`.

**Comment by angersson** [comment]
@ebrevdo, can you comment on this?

**Comment by ebrevdo** [comment]
@petewarden do you have experience with dynamic rnn and benchmark_model?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @ebrevdo: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @ebrevdo: It has been 52 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @ebrevdo: It has been 67 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @ebrevdo: It has been 82 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @ebrevdo: It has been 97 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatianashp: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by songmeixu** [comment]
I am facing the same issue, look like `dynamic rnn` is a problem for `benchmark_model`. Any new progress?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatianashp: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatianashp: It has been 30 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by shqking** [comment]
I encountered the same issue, trying to utilize benchmark_model to evaluate the FLOPS for my custom RNN model.

**Comment by lissyx** [comment]
We're moving away from `bidirectional_dynamic_rnn` :/

**Comment by shqking** [comment]
@lissyx

**Comment by lissyx** [comment]
Thanks, this was not documented anywhere when we started that.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatianashp: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatianashp: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatianashp: It has been 44 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatianashp: It has been 60 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @tatianashp: It has been 78 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by mohantym** [comment]
Hi @lissyx ! Could you try with [keras-flops ](https://pypi.org/project/keras-flops/)(external contributor) with keras.layers.Bidirectional(keras.layers.RNN(cell) (Alternate of tf.compat.v1.nn.bidirectional_dynamic_rnn in 2.x version  )
